specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2019-10-01/componentFeaturesAndPricing_API.json

```diff
@@ -0,0 +1,484 @@
+            }
+          }
+        },
+        "x-ms-examples": {
+          "ComponentCurrentBillingFeaturesGet": {
+            "$ref": "./examples/AvailableBillingFeaturesGet.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ApplicationInsightsComponentBillingFeatures": {
+      "type": "object",
+      "description": "An Application Insights component billing features",
+      "properties": {
+        "DataVolumeCap": {
+          "description": "An Application Insights component daily data volume cap",
+          "$ref": "#/definitions/ApplicationInsightsComponentDataVolumeCap"
+        },
+        "CurrentBillingFeatures": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "Current enabled pricing plan. When the component is in the Enterprise plan, this will list both 'Basic' and 'Application Insights Enterprise'."
+        }
+      }
+    },
+    "ApplicationInsightsComponentDataVolumeCap": {
+      "type": "object",
+      "description": "An Application Insights component daily data volume cap",
+      "properties": {
+        "Cap": {
+          "type": "number",
+          "description": "Daily data volume cap in GB."
+        },
+        "ResetTime": {
+          "type": "integer",
+          "readOnly": true,
+          "description": "Daily data volume cap UTC reset hour."
+        },
+        "WarningThreshold": {
+          "type": "integer",
+          "description": "Reserved, not used for now."
+        },
+        "StopSendNotificationWhenHitThreshold": {
+          "type": "boolean",
+          "description": "Reserved, not used for now."
+        },
+        "StopSendNotificationWhenHitCap": {
+          "type": "boolean",
+          "description": "Do not send a notification email when the daily data volume cap is met."
+        },
+        "MaxHistoryCap": {
+          "type": "number",
+          "readOnly": true,
+          "description": "Maximum daily data volume cap that the user can set for this component."
+        }
+      }
+    },
+    "ApplicationInsightsComponentQuotaStatus": {
+      "type": "object",
+      "description": "An Application Insights component daily data volume cap status",
+      "properties": {
+        "AppId": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The Application ID for the Application Insights component."
+        },
+        "ShouldBeThrottled": {
+          "type": "boolean",
+          "readOnly": true,
+          "description": "The daily data volume cap is met, and data ingestion will be stopped."
+        },
+        "ExpirationTime": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Date and time when the daily data volume cap will be reset, and data ingestion will resume."
+        }
+      }
+    },
+    "ApplicationInsightsComponentFeatureCapabilities": {
+      "type": "object",
```

Comment:
Throughout this PR, there are many noncompliant request/response schemas. Please familiarize with the contract: https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md